### PR TITLE
[terra-clinical-detail-view] Add aria-labelledby to section and associate to the title, update example in dev site with graph

### PR DESCRIPTION
**Pull request by kolkheang**
### Summary

Provide a better experience for screen reader users when navigating the detail view section

**What was changed:**
- Add aria-labelledby to `<section>` and associate to the primary title of DetailView
- Update an example in dev site to include terra-graphs example

**Why it was changed:**
- Provide more context to screenreader users when navigating the detail view
- Provide a realistic example of using terra-graphs for the `graph` prop

### Testing

This change was tested using:

- [ ] WDIO
- [x] Jest
- [x] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [x] UX review
- [x] Accessibility review
- [ ] Functional review

### Additional Details

**VO for section and title**

https://github.com/cerner/terra-clinical/assets/3170314/f335d88b-b8a2-4d27-9a4d-608ade0a359d

**VO for graph example**

https://github.com/cerner/terra-clinical/assets/3170314/726504e2-e510-4b8b-8c64-297e30da78ec

**This PR resolves:**

UXPLATFORM-9150 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Comment by kolkheang** [comment]
Dev Site updated in this PR 
https://engineering.cerner.com/terra-clinical/pull/869/

**Comment by kolkheang** [comment]
FYI - @chrismichalewicz

**Review by brooksienoodlesoup — APPROVED** [review]
Tested in:

- Chrome 108.0.5359.94 + Voice Over (Ventura 13.2.1)
- Safari 16.3 (18614.4.6.1.6) + Voice Over (Ventura 13.2.1)
- Edge 114.0.1823.43 + JAWS 2023.2303.144
Works as expected.  section/region was identified and appropriately named in all three browser/screen reader combinations.

**Comment by chrismichalewicz** [comment]
@kolkheang  This looked good from a product perspective.

I also updated the Accessibility and UX labels as it appears like @brooksienoodlesoup reviewed and approved it.

File: packages/terra-clinical-detail-view/CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,11 @@
 ## Unreleased
 * Changed
   * Updated the component examples.
-  
+
+* Added
+  * Added aria-labelledby attribute to DetailView `<section>` using title's id attribute.
+  * Added an example terra-graphs for `graph` prop to terra-clinical-detail-view dev site
```
Comment (RayGunY) [inline packages/terra-clinical-detail-view/CHANGELOG.md:9]: Nit: Add a period to this line

**Comment by RayGunY** [comment]
Hey Kol, this isn’t anything you did but if you have the time would you mind updating the DetailViewDivided file? The text it’s using to populate the graph is off. The two comments (detail1item3 and detail2item3) don’t have correct spacing/some spelling errors: https://github.com/cerner/terra-clinical/blob/main/packages/terra-clinical-detail-view/src/terra-dev-site/doc/example/DetailViewDivided.jsx#L12 

First one should be: “Based on the patient's blood glucose levels and medical history, I recommend that they continue to monitor their glucose levels regularly and adhere to a healthy diet and exercise regimen. It's important for the patient to attend regular follow-up appointments and communicate any concerns in symptoms.”

The second one should be: “Based on the patient's blood pressure readings and medical history, I recommend that they make lifestyle modifications such as reducing their salt intake, increasing physical activity, and losing weight if necessary. It's important for the patient to continue monitoring their blood pressure at home and attending regular follow-up appointment to assess the effectiveness of their treatment plan”

Everything else looks great - that graph is beautiful!

File: packages/terra-clinical-detail-view/CHANGELOG.md [hunk 2]
```diff
@@ -3,7 +3,11 @@
 ## Unreleased
 * Changed
   * Updated the component examples.
-  
+
+* Added
```
Comment (zxeleven) [inline packages/terra-clinical-detail-view/CHANGELOG.md:7]: nit: from Vinay's comment on my PR, I think these should be in the Added section on line 11

**Review by zxeleven — APPROVED** [review]

**Review by RayGunY — APPROVED** [review]

**Comment by kolkheang** [comment]
@sdadn  / @sycombs  could you take a look when you get a chance?

**Review by sdadn — CHANGES_REQUESTED** [review]

File: packages/terra-clinical-detail-view/package.json [hunk 3]
```diff
@@ -30,7 +30,8 @@
     "prop-types": "^15.5.8",
     "terra-clinical-label-value-view": "^3.28.0",
     "terra-icon": "^3.52.0",
-    "terra-theme-context": "^1.0.0"
+    "terra-theme-context": "^1.0.0",
+    "uuid": "^8.3.2"
```
[the remark this answers is not in the document]
Comment (kolkheang) [reply packages/terra-clinical-detail-view/package.json]: Fixed with: https://github.com/cerner/terra-clinical/pull/869/files#diff-c2b00b078c52adb67098ef3ab5c20d7bb1a1a95f4d51c993b6b048da8ed03ef3R34

File: packages/terra-clinical-detail-view/src/DetailView.jsx [hunk 4]
```diff
@@ -13,6 +14,9 @@ const cx = classNamesBind.bind(styles);
 const propTypes = {
   /**
    * Sets the text to display for the main heading.
+   *
+   * ![IMPORTANT](https://badgen.net/badge/UX/Accessibility/blue) It is critical to screen reader users that the title prop be set to a value that best represents the region which groups the details in section within in the page.
+   * Screen reader users rely on the title to know the begin and end of a region, and to navigate to the specific region on the page.
```
[the remark this answers is not in the document]
Comment (kolkheang) [reply packages/terra-clinical-detail-view/src/DetailView.jsx]: Fixed with: https://github.com/cerner/terra-clinical/pull/869/files#diff-7c93559c8fcc0937cc491d416ca9b67c7be5123cca0eed7b04d55d11652c5e9dR19

File: packages/terra-clinical-detail-view/tests/jest/DetailView.test.jsx [hunk 5]
```diff
@@ -1,6 +1,8 @@
 import React from 'react';
 import DetailView from '../../src/DetailView';
 
+jest.mock('uuid', () => ({ v4: () => 'dacd01de-e7d0-4e1a-a8f1-eba662631b7d' }));
```
[the remark this answers is not in the document]
Comment (kolkheang) [reply packages/terra-clinical-detail-view/tests/jest/DetailView.test.jsx]: Fixed with https://github.com/cerner/terra-clinical/pull/869/files#diff-cbf723360a8e7bc0de6c6c28d8f6b3a65156666904f3b54b8a724cbe75bf37cfR4

File: packages/terra-clinical-detail-view/CHANGELOG.md [hunk 6]
```diff
@@ -2,15 +2,18 @@
...
+
 ## 3.30.0 - (June 14, 2023)
 
 * Changed
-  * Updated the component examples.
   * Updated the DetailView component to use semantic html elements.
 
 * Added
   * Added `level` prop to the DetailView component to allow users to render different heading levels.
-  * Added `isLabelValuePairList` prop to the DetailList component to support using a description list for label-value pairs.
```
Comment (kolkheang) [inline packages/terra-clinical-detail-view/CHANGELOG.md:13]: @zxeleven  - FYI, I move this up to the un-released section

Comment (zxeleven) [reply packages/terra-clinical-detail-view/CHANGELOG.md:13]: Thanks!

**Comment by kolkheang** [comment]
@sdadn - I addressed your comments.

**Comment by kolkheang** [comment]
I'm going make one small update to remove this text "This area is where graph, image, visualization content, and text can optionally be displayed." from the Multiple Ingredients example since we have a better graph example in the Mother example.

Screenshot 2023-06-21 at 10 09 58 AM

**Review by RayGunY — APPROVED** [review]

**Review by zxeleven — APPROVED** [review]

**Review by sdadn — APPROVED** [review]

**Pull request merged**